ptf

```diff
@@ -626,9 +626,16 @@ if config["list_test_names"]:
 
 # Generate the test suite
 test_suite = []
+tests_executed = []
+test_report = []
```

Comment:
There doesn't seem to be a need for both `tests_executed` and `test_report`.

It also seems that you should be using a dictionary (mapping test name to test information, such as status), based on the usage you do of this data